### PR TITLE
Initial team service implementation

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary

This is the first part of team service implementation. I started with moving some basic parts as Create/Update/Patch etc. Also included some `channel` and `user` entity relation to get some early feedback so that we can discuss how the relation will be between different services in the future.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/teams/teams.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	displayNames := map[string]string{
+		"town-square": i18n.T("api.channel.create_default_channels.town_square"),
+		"off-topic":   i18n.T("api.channel.create_default_channels.off_topic"),
+	}
+	channels := []*model.Channel{}
+	defaultChannelNames := ts.DefaultChannelNames()
+	for _, name := range defaultChannelNames {
+		displayName := i18n.TDefault(displayNames[name], name)
+		channel := &model.Channel{DisplayName: displayName, Name: name, Type: model.ChannelTypeOpen, TeamId: teamID}
+		// We should use the channel service here (coming soon). Ideally, we should just emit an event
```
Comment (agnivade) [inline app/teams/teams.go:48]: I was of the understanding that we would keep all "Channel" related stuff inside app and move general stuff which are not tied to channels outside. Would there be a need to move channels outside as well?

**Comment by agnivade** [comment]
Looks like a good start, but I want to gather some thoughts and have a plan before we make more changes.

@jespino , @lieut-data - Are there any requirements from focalboard/playbooks to access team related data? What is the user grouping level your products are looking to have? Does everything organize under a team, or will there be a need for custom user grouping as well?

From my conversations with @jwilander - IIRC, focalboard was looking for custom user groups support apart from LDAP.

If that's the case, I am wondering if "teams" should remain as a channels concept or a global entity?

This ties in with the unified suite architecture that we are looking to have - where channels, boards and playbooks are entities that deeply integrate with the `App` struct. We have already moved out users from `App`. But we should only move out stuff that will be used by other teams.

File: app/teams/teams.go (hunk 1 above)
Comment (isacikgoz) [reply app/teams/teams.go:48]: In the end picture we may have a channels package which will be responsible of channel related stuff. The idea here is to make a connection between team and channels. Let's say we want to create a team from Boards path, we should just ask teams package to create the team then it should notify/call required entities and complete the process.

**Comment by jwilander** [comment]
Based on our discussion yesterday I think "teams" should definitely still be a global entity instead of just a channels one. Playbooks still uses teams and Boards will too. In the long term future what teams are will likely evolve but I don't think they're going to away or be something that only one product uses.

**Comment by isacikgoz** [comment]
@agnivade @streamer45 gentle ping guys :) I want to work on remaining parts after merging this one.

**Comment by agnivade** [comment]
Woops, this got out of my queue. Will review by EOD.

File: app/channel.go [hunk 2]
```diff
@@ -2337,7 +2317,17 @@ func (a *App) removeUserFromChannel(c *request.Context, userIDToRemove string, r
...
+			message := model.NewWebSocketEvent(model.WebsocketEventLeaveTeam, teamMember.TeamId, "", "", nil)
+			message.Add("user_id", teamMember.UserId)
+			message.Add("team_id", teamMember.TeamId)
+			a.Publish(message)
+
+			if err := a.srv.teamService.RemoveTeamMember(teamMember); err != nil {
+				return model.NewAppError("removeUserFromChannel", "api.team.remove_user_from_team.missing.app_error", nil, err.Error(), http.StatusBadRequest)
+			}
+
+			if err = a.postProcessTeamMemberLeave(c, teamMember, removerUserId); err != nil {
```
Comment (agnivade) [inline app/channel.go:2330]: Looks like this logic is duplicated in app/channel.go and app/team.go. Can we refactor this to a common method?

Comment (isacikgoz) [reply app/channel.go:2330]: Sure 👍

**Pull request merged**